modules/tinymce/src/plugins/image/main/ts/ui/Dialog.ts

```diff
@@ -174,12 +174,15 @@ const calculateImageSize = (helpers: Helpers, info: ImageDialogInfo, state: Imag
   const data = api.getData();
   const url = data.src.value;
   const meta = data.src.meta || {};
-  if (!meta.width && !meta.height && info.hasDimensions) {
+
+  if (!meta.width && !meta.height && info.hasDimensions && Strings.isNotEmpty(url)) {
```

Comment:
This doesn't look right... We only want to clear the width and height if we're supposed to be getting the dimensions and the URL is empty. At the moment it'll clear it under various other conditions as well (eg if only the width, but not the height is set).

- [ ] Fix and make sure to add a test case

Reply:
Done, minus the test :-)
Any ideas on that more than appreciated

Reply:
@metricjs, @lnewson I've added your suggestion to the test. Could you please review on DialogUpdateTest.ts?

Reply:
uhmmm wait... I've only added: 
file_picker_callback(callback, _value, _meta) {
  callback('https://www.google.com/logos/google.jpg', { width: '100' });
},

but actually missed this:
" that should make sure only a width is set. When I tested that using your old code, the dialog had no width/height values.
so we need to set the above setting, click the "file picker" button to the right of the source url
and then assert the width in the dialog and what gets inserted into the editor keeps the width value "

Reply:
Done. @metricjs @lnewson . Thanks for the help @JamesToohey